### ISSUE TITLE
Rasterizer backward is too slow

### ISSUE THREAD

**Issue by Master-cai** [RaymondJiangkw/COGS#13]
Hi! Recently, I have apply your diff-gaussian-rasterization-appr-surface to a vanilla gaussian model, and use `DifferentiableCamera` for pose optimization. The initial point cloud has 880k points, the loss is the same loss as vanilla gaussian (l1 + dssim). However, I find the loss.backward() takes too much time, around 1.9s per step. And i find that in your demo, backward of 500k points only takes 50ms and 1m points takes 100ms. Do you have any ideas? Thanks!
<img width="491" alt="image" src="https://github.com/user-attachments/assets/ca9c5c75-f913-4d80-9fb5-16733a24fe98">

cc @RaymondJiangkw

**Comment by RaymondJiangkw** [comment]
Hi, that is due to the size of Gaussian kernels. For the demo, the size is pretty small and therefore it is fast. But for the regular initialized Gaussian kernels for view synthesis, it will slow down.

I recently have tested the gsplat, which also supports extrinsics differentiation now. It seems to work nice and fast. You can also tried it out. :)
